### PR TITLE
Update documentation for mx.callback.do_checkpoint

### PR DESCRIPTION
**Pull request by indhub**
@mli, @zackchase, @madjam, @nswamy, @Roshrini, @jiajiechen
Please review and merge.

### CODE REVIEW

File: python/mxnet/callback.py [hunk 1]
```diff
@@ -37,19 +37,34 @@ def _callback(iter_no, sym=None, arg=None, aux=None):
 
 
 def do_checkpoint(prefix, period=1):
-    """Callback to checkpoint the model to prefix every epoch.
+    """A callback that saves a model checkpoint every few epochs. Checkpoint is binary files
```
Comment (madjam) [inline python/mxnet/callback.py]: Suggest rewording as:

A callback that saves a model checkpoint every few epochs. Each checkpoint is made up of couple of binary files: a model description file and a parameters (weights and biases) file. The model description file is named `prefix`--symbol.json and parameters file is named `prefix`-`epoch_number`.params

Comment (zackchase) [inline python/mxnet/callback.py]: Thanks madan for catching this. 

A few changes to your changes: 
* "made up of *a* couple of ..."
* "and the parameters file is named ..."

File: python/mxnet/callback.py [hunk 2]
```diff
@@ -37,19 +37,34 @@ def _callback(iter_no, sym=None, arg=None, aux=None):
...
 
     Returns
     -------
     callback : function
-        The callback function that can be passed as ``iter_end_callback`` to fit.
+        A callback function that can be passed as `epoch_end_callback` to fit.
+
+    Example
+    -------
+    >>> model.fit(X=trainingX, y=trainingY,
```
Comment (piiswrong) [inline python/mxnet/callback.py]: use the module interface. Not model
